core/src/main/java/io/questdb/cairo/mig/Mig614.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.mig;
+
+import io.questdb.cairo.vm.MemoryCMARWImpl;
+import io.questdb.cairo.vm.api.MemoryMARW;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.MemoryTag;
+import io.questdb.std.str.Path;
+
+import static io.questdb.cairo.TableUtils.*;
+
+final class Mig614 {
+    private static final long TX_OFFSET_STRUCT_VERSION = 40;
+    private static final long META_OFFSET_STRUCTURE_VERSION = 32;
+    private static final Log LOG = LogFactory.getLog(EngineMigration.class);
+
+    static void migrate(MigrationContext migrationContext) {
+        final FilesFacade ff = migrationContext.getFf();
+        final Path path = migrationContext.getTablePath();
+
+        try (MemoryMARW metaMem = migrationContext.getRwMemory()) {
+            metaMem.of(ff, path.concat(META_FILE_NAME), ff.getPageSize(), ff.length(path), MemoryTag.NATIVE_DEFAULT);
+
+            try (MemoryMARW txMem = new MemoryCMARWImpl(
+                    ff,
+                    migrationContext.getTablePath2().concat(TXN_FILE_NAME).$(),
+                    ff.getPageSize(),
+                    ff.length(path),
```

Comment:
same.

This is a very minor spaghetti, I would perhaps read _txn cleanly, close it, then open _meta and write value.

Reply:
changed in 717531a